nulib/react-media-player
# Support VTT nested chapters, and updates handling current time as a React context

### Description
This PR:

- Set up webVtt hook and handle initial parsing of flat structure into nested cues structure
- Update to complete nested structure from a flat array.  Extends beyond one child.
- First pass at nested UI display
- Update styling for nested UI display
- Logging code to track down why the Radix Tab Content component is blocking a new Resource from displaying, when changing active canvas.
- Fix potential orphan childed if parent end times are equal.
- Remove background styling on nested children.
- Remove console logs.

![image](https://user-images.githubusercontent.com/3020266/149586242-28bf18c4-93c6-4488-8e0d-b88a90c4b284.png)

It also pulls out the state variable keeping track of the Video player's `currentTime`, and puts it into its own Context.  This reduces top-level component renders, and moves the continuous rendering out to primarily the `Cue` component.  Somewhat of a performance optimization to be realized if WebVTT files get huge.

### To Test
In the current PR Code, click on the 4th Video in the Tray ("Video canvas, non-streaming (.mp4), with supplementing annotations as WebVTT"), to see nested navigation in the UI. 

You'll also notice in the React Dev Tools Profiler tab, a reduced rendering time, who's Flamegraph used to look like:

![image](https://user-images.githubusercontent.com/3020266/149587244-e88cb701-a670-4224-b26c-439ed92d445e.png)

and now looks much more like:

![image](https://user-images.githubusercontent.com/3020266/149587339-7d1c5213-f175-4de2-b80c-20100c046181.png)

Pushing all the rendering down to really quick renders.
